state/executor.go

```diff
@@ -137,10 +137,6 @@ func (e *Executor) ProcessBlock(
 
 	for _, t := range block.Transactions {
 		if t.ExceedsBlockGasLimit(block.Header.GasLimit) {
-			if err := txn.WriteFailedReceipt(t); err != nil {
```

Comment:
Worth considering do we even need this check in this place?

```go
if t.ExceedsBlockGasLimit(block.Header.GasLimit)
```

Reply:
Maybe someone malicious built block with larger block gas limit then it should be?

Reply:
Ok, but it is already being checked when building a block itself, so not sure do we need a check here as well. If we are about to leave it, then maybe consider adding a log.

Reply:
@Stefan-Ethernal fixed: https://github.com/0xPolygon/polygon-edge/pull/1403/commits/615b8f04cedb9c0ec24700600f5a6ac4aa829a49